libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1164,6 +1165,19 @@ blk_hash_t PbftManager::calculateOrderHash(std::vector<DagBlock> const &dag_bloc
   return dev::sha3(order_stream.out());
 }
 
+std::optional<blk_hash_t> findClosestAnchor(const std::vector<blk_hash_t> &ghost, std::vector<blk_hash_t> &dag_order,
+                                            uint32_t included) {
+  for (uint32_t i = included; i > 0; i--) {
+    if (std::count(ghost.begin(), ghost.end(), dag_order[i - 1])) {
+      if (dag_order.size() > i) {
+        dag_order.erase(dag_order.begin() + i, dag_order.end());
+      }
+      return dag_order[i - 1];
+    }
+  }
+  return {};
```

Comment:
I think we should return ghost[0] here. And also need modify checkBlockWeight return true if weight < gas limit OR DAG anchor is ghost[0]

Reply:
If you modify this way, I think you cannot verify synced PBFT block when node just receive it because DAG blocks have not exist at the moment. That need to change to verify just before adding to PBFT chain